### ISSUE TITLE
Table functions `generate_series`, `generate_date_array`, `generate_timestamp_array`

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#47041]
**Use case**

Compatibility with Postgres and BigQuery.

**Describe the solution you'd like**

We already have the `numbers` table function, but we can add these functions for convenience.

**Additional context**

https://www.postgresql.org/docs/current/functions-srf.html

**Comment by alexey-milovidov** [comment]
@zhujing0227 yes, it makes sense to extend the `range` function to support Date, DateTime, Date32 and DateTime64:

`range(date1, date2)` - will generate dates with one day step by default;
`range(time1, time2)` - will generate date-times with one-second step by default;
`range(time1, time2, INTERVAL ...)` - will generate a range with a specified step interval.

**Comment by iiiuwioajdks** [comment]
I want to try to implement it, can i know more about this problem

**Comment by Ziy1-Tan** [comment]
take

**Comment by hillbilly991** [comment]
was there any update on this? I am trying to implement something like this now

**Comment by hp77-creator** [comment]
@alexey-milovidov so extending range should be the way or would you prefer new functions like how its in BigQuery?
